tests/test_who.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+const test = require('tape');
+const mocks = require('./mocks');
+
+test('irc_who', async(t) => {
+  t.plan(3 + mocks.connectOneIrcClient.planCount);
+  const c = await mocks.connectOneIrcClient(t);
+  c.ircUser.mapIrcToSlack('fun_user', 'U1234USER');
+  c.slackWeb.expect('users.list', {
+    presence: false,
+    limit: 1000}, {
+    ok: true,
+    members: [
+      { name: 'fun_user',
+        id: 'U1234USER',
+        profile: [
+          {email: 'foo@example.com',
+            real_name: 'Foo Bar' },
+        ]},
+    ],
+  });
+  c.ircSocket.expect(':irslackd 352 test_slack_user * U1234USER irslackd example.com fun_user nobody@example.com 0 Nobody');
+  c.ircSocket.expect(':irslackd 315 test_slack_user :End of WHO');
```

Comment:
Below is a packet capture of a `WHO` response from an undernet server:

```
:Montreal.QC.CA.Undernet.Org 352 adam1 * adam XYZ.verizon.net *.undernet.org adam1 H :0 adam
:Montreal.QC.CA.Undernet.Org 315 adam1 adam :End of /WHO list.
```

To me it seems like the format should be:

```
:irslackd 352 <our_nick> * <their_nick> irslackd irslackd <our_nick> H :0 <their_name>
```

or in our case:

```
:irslackd 352 test_slack_user * fun_user irslackd irslackd test_slack_user H :0 Foo Bar
```